### PR TITLE
[Feature][Branch][Select] Implement Picker mode on Select Component

### PR DESCRIPTION
**Pull request by dlgmltjr0925**
## Description

Implement Picker mode on Select Component
1. position: Implemented to show center of list based on root.
2. style: It provides basic styles based on themes and makes it possible to customize the styles of list containers and items according to the user's purpose.
3. select: When scrolling through the list, the value in the middle of the list was changed by calling the [onValueChange] callback.
You can also click on a specific item to change its value.
4. Redefining the text of an [Item] interface with a label seems more intuitive.
5. Added information about Picker to the README.
-TODO : Will be cleaned up after another list mode has been added.

<div>
    picker-ios-demo
    picker-android-demo
</div>

## Related Issues

#60 Picker of [Select] Component

## Tests

```
Interaction when mode is picker
  Check the interaction of the theme and style
    ✓ Make sure that none is applied when the theme is undefined (16ms)
    ✓ Make sure that box is applied when the theme is "box" (14ms)
    ✓ Make sure that default is applied when the theme is "blank" (12ms)
    ✓ Make sure that default is applied when the theme is "underbar" (10ms)
  Check the interaction of the custom style
    ✓ Check that the top and left styles disappear 
  when you apply bottom and right styles to the itemListStyle (11ms)
    ✓ Check that the top and left styles disappear 
  when you apply bottom and right styles to the itemStyle (19ms)
    ✓ If bottom and right are not applied to itemListStyle, the default value is applied (12ms)
  Check the interaction based on user behavior
    ✓ If onValueChange is not defined, no callback is executed (20ms)
    ✓ This list disappeaers when you press a part other than the list (16ms)
    ✓ Pressing on an item that is not selected will move to scroll (10ms)
    ✓ called callback function when move to scroll (34ms)
  Check the interaction other props
    ✓ Given a selected value, the initial position of the list is determined (12ms)
```

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui-native/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: src/components/shared/Select/index.tsx [hunk 1]
```diff
@@ -221,53 +246,84 @@ const RootSelect = styled.View<ThemeType>`
...
       : defaultTheme;
   const rootTextTheme = disabled
-    ? 'disabled'
+    ? ThemeEnum.disabled
     : rootTextStyle && Object.keys(rootTextStyle).length > 0
-      ? 'blank'
+      ? ThemeEnum.blank
       : defaultTheme;
   const titleTextTheme = disabled
     ? 'disabled'
```
Comment (geoseong) [inline src/components/shared/Select/index.tsx]: I think this string should be replaced to`ThemeEnum.disabled` as you changed at previous `rootViewTheme` definition line.

File: src/components/shared/Select/index.tsx [hunk 2]
```diff
@@ -221,53 +246,84 @@ const RootSelect = styled.View<ThemeType>`
...
-    : titleTextStyle && Object.keys(titleTextStyle).length > 0
+    : titleStyle && Object.keys(titleStyle).length > 0
       ? 'blank'
```
Comment (geoseong) [inline src/components/shared/Select/index.tsx]: this should be replaced to `ThemeEnum.blank` also.

File: src/components/shared/Select/index.tsx [hunk 3]
```diff
@@ -301,25 +358,40 @@ function Select(props: Props): React.ReactElement {
...
+        <View style={{ height: 50, width: 100, backgroundColor: 'white' }} />
+      )}
+      {/* Todo dialog */}
+      {mode === Mode.dialog && listOpen.isOpen && (
+        <View style={{ height: 50, width: 100, backgroundColor: 'white' }} />
+      )}
+      {mode === Mode.picker && (
+        <Picker
+          {...props}
+          testID={'select-picker'}
```
Comment (geoseong) [inline src/components/shared/Select/index.tsx]: I think `testID` value should be managed like `${testID}-${TESTID.PICKER}`

**Review by geoseong — CHANGES_REQUESTED** [review]
When I try to expand picker items, only white little box appeared. can you explain how can I make this work?

스크린샷 2020-01-07 오전 1 03 45

**Comment by dlgmltjr0925** [comment]
When I try to expand picker items, only white little box appeared. can you explain how can I make this work?

Currently the props mode is set to dropdown mode. Enter mode as "picker" and try.

File: src/components/shared/Select/index.tsx (hunk 3 above)
Comment (dlgmltjr0925) [reply src/components/shared/Select/index.tsx]: Thank you for letting me know what I missed while writing the test code.

File: src/components/shared/Select/index.tsx (hunk 1 above)
Comment (dlgmltjr0925) [reply src/components/shared/Select/index.tsx]: According to your comment, I changed the string part to ThemeEnum type.
It looks much better.

File: src/components/shared/Select/index.tsx (hunk 2 above)
Comment (dlgmltjr0925) [reply src/components/shared/Select/index.tsx]: I changed this too.

**Comment by geoseong** [comment]
> When I try to expand picker items, only white little box appeared. can you explain how can I make this work?
> 
> Currently the props mode is set to dropdown mode. Enter mode as "picker" and try.

Okay. It works!

and I have a question the concept of `Picker`.
If I select one item in `Picker`, I expect `Picker` component disappeared and shown selected item on `root` area. but when I select one item in `Picker`, just focused to selected item and `Picker` component doesn't disappeared.
Is this concept intentional?

**Comment by dlgmltjr0925** [comment]
My intended Picker was to change the selectedValue by executing an onChangeValue callback when the item was focused while scrolling. So I thought it was correct that the list would not disappear until the desired value was selected. However, I wondered if the list should disappear when I press the selected value one more time.

**Comment by geoseong** [comment]
Well, I think once item selected after scrolling, It would be nice to click item one more time to make picker disappeared.
or make `decide` button in corner of **Picker** like iOS's Picker style.
What do you think?
You can decide yourself if your thought of Picker's concept is good enough.

**Review by geoseong — APPROVED** [review]

**Comment by dlgmltjr0925** [comment]
@geoseong Fixed the screen disappearing when clicking on a selected item of your opinion.

**Pull request merged**